File: external/HDF-EOS/hdfeos/samples/subsetgrid.f
c
c  In this example we will (1) open the "GridFile" HDF file, (2) attach to
c  the "PolarGrid" grid, and (3) subset data from the "Temperature" field.
c 


	program subsetgrid

	integer            status, gdextreg, gdreginfo, gddetach, gdclose
	integer*4          gdfid, gdid, gdopen, gdattach, dims(8)
	integer*4          rank, ntype, gddefboxreg

	real*8             cornerlon(2), cornerlat(2)
	real*8             upleft(2), lowright(2)
	real*4             datbuf(100*100)
	
	integer DFACC_READ
	parameter (DFACC_READ=1)


c    
c     Open the HDF grid file, "GridFile.hdf"
c 

	gdfid = gdopen("GridFile.hdf", DFACC_READ)

	if (gdfid .NE. -1) then

	   gdid = gdattach(gdfid, "PolarGrid")

	   if (gdid .NE. -1) then

	      cornerlon(1) = 0
	      cornerlat(1) = 90.
	      cornerlon(2) = 90
	      cornerlat(2) = 0

	      regionid = gddefboxreg(gdid, cornerlon, cornerlat)
	      
	      status = gdreginfo(gdid, regionid, "Temperature", ntype,
     1	                         rank, dims, size, upleft, lowright)
	      write(*,*) dims(1), dims(2), dims(3), rank, ntype
	      
	      status = gdextreg(gdid, regionid, "Temperature", 
     1	                        datbuf)
	    

	   endif
	
	   status = gddetach(gdid)
	   status = gdclose(gdfid)

	endif
	stop
	end
	

File: external/HDF-EOS/hdfeos/samples/SubsetGrid.c
#include "hdf.h"
#include <math.h>

/*
 * In this example we will (1) open the "GridFile" HDF file, (2) attach to
 * the "PolarGrid", and (3) subset data from the "Temperature" field.
 */


main()
{

    intn            i, j, status;
    
    int32           gdfid, GDid, regionID, size, dims[8], ntype, rank;

    float32        *datbuf32;
    
    float64         cornerlon[2], cornerlat[2];
    float64        *datbuf64, upleft[2], lowright[2];
    


    /*
     * Open the HDF grid file, "GridFile.hdf".
     */

    gdfid = GDopen("GridFile.hdf", DFACC_RDWR);


    if (gdfid != -1)
    {

	GDid = GDattach(gdfid, "PolarGrid");

	if (GDid != -1)
	{
	    cornerlon[0] = 57.;
	    cornerlat[0] = 23.;
	    cornerlon[1] = 59.;
	    cornerlat[1] = 35.;

	    cornerlon[0] = 0.;
	    cornerlat[0] = 90.;
	    cornerlon[1] = 90.;
	    cornerlat[1] = 0.;


	    regionID = GDdefboxregion(GDid, cornerlon, cornerlat);
	    
	    status = GDregioninfo(GDid, regionID, "Temperature", &ntype,
				&rank, dims, &size, upleft, lowright);
	    printf("size: %d\n",size);
	    
	    datbuf32 = (float32 *) calloc(size, 1);
	    
	    status = GDextractregion(GDid, regionID, "Temperature", 
				     datbuf32);
	    
	    free(datbuf32);
	}
    }

    GDdetach(GDid);

    GDclose(gdfid);

    return;
}
